### ISSUE TITLE
Add support for CancellationToken in AuthorizationHandler or AuthorizationHandlerContext

### ISSUE THREAD

**Issue by etiennelepagel** [dotnet/aspnetcore#56272]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

We would need a CancellationToken in our AuthorizationHandlers.

### Describe the solution you'd like

As discussed in previous issues (https://github.com/dotnet/aspnetcore/issues/2746 & https://github.com/aspnet/Security/issues/1598) a non-breaking way of doing this would be to add the CancellationToken in the AuthorizationHandlerContext).

**Comment by etiennelepagel** [comment]
> > A work around we have for now it to get the ctx from the HttpContextAccessor.
> 
> You can also use `(AuthorizationHandlerContext.Resource as HttpContext).RequestAborted` for this in the meantime.
> 
> https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/5.0/authorization-resource-in-endpoint-routing

@halter73 

Why not expose the entire HttpContext explicitly in the AuthorizationHandlerContext? Is the AuthorizationHandler ever used outside of an HTTP request context?

**Comment by halter73** [comment]
Yes. One example is Blazor WebAssembly. Microsoft.AspNetCore.Authorization does not have a dependency on Microsoft.AspNetCore.Http.Abstractions.

It would be more feasible to add a CancellationToken to AuthorizationHandlerContext like you suggest. If not for the Microsoft.AspNetCore.Authorization NuGet package which supports netstandard2.0 and net462, we could also consider a new IAuthorizationHandler.HandleAsync overload that takes a CancellationToken parameter directly with a [default implementation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods) that drops it.

It's not a high priority given the workaround, but we recognize that it's not discoverable.

**Comment by WORMrus** [comment]
As a user of [Hot Chocolate](https://github.com/ChilliCream/graphql-platform) (one of the big C# libs for GraphQL servers), I would like to add that they are implementing their auth mechanism at least in part on the pre-existing types (such as `AuthorizationHandler`s). 

They do provide `AuthorizationHandlerContext.Resource` and it does have a token, but the underlying type is not an `HttpContext`.

This is a problem because it somewhat complicates writing extension libraries targeting both ASP and HC apps as you need to know the "execution context" (ASP\HC) where you would really like not to. 

The `HttpContextAccessor` works, but the token semantics could be somewhat different between this one and the native one from `AuthorizationHandlerContext.Resource`. 

So having the dedicated "platform-agnostic" place to get the tokens would be great.
